### PR TITLE
[IA-4860][IA-4593] Delete and Create azure app polling consolidation 

### PR DESCRIPTION
**Pull request by jdcanas**

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala [hunk 1]
```diff
@@ -1083,8 +1076,8 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
           case Some(identity) =>
             logger.info(
               s"Managed ID found in WSM app ${app.appName}, using previously created identity: ${identity.managedIdentityName}"
-            )
-            F.pure(Option(identity))
+            ) >>
```
Comment (jdcanas) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala:1074]: 😬

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala [hunk 2]
```diff
@@ -1295,26 +1286,80 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
       )
       result <- streamFUntilDone(
         op,
-        config.appMonitorConfig.createApp.maxAttempts,
-        config.appMonitorConfig.createApp.interval
```
Comment (jdcanas) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala:1299]: ©️ 🍝

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala [hunk 3]
```diff
@@ -1453,6 +1453,7 @@ class LeoPubsubMessageSubscriber[F[_]](
     e match {
       case e: PubsubKubernetesError =>
         for {
+          _ <- logger.error(ctx.loggingCtx, e)("top level handler for handleKubernetesError")
```
Comment (jdcanas) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala:1469]: I think its worth explicitly logging here to make sure every app error gets logged to console, not just saved to db

File: .github/workflows/leo-build-tag-publish-and-run-tests.yml [hunk 4]
```diff
@@ -222,6 +222,8 @@ jobs:
           repo: DataBiosphere/leonardo
           ref: refs/heads/develop
           token: ${{ env.TOKEN }}
+          # this is not actually 85mins, the timeout is not working correctly and is summing from other jobs
```
Comment (LizBaldo) [inline .github/workflows/leo-build-tag-publish-and-run-tests.yml:225]: Yeah I am going to post our findings and ask Devops this afternoon

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala [hunk 5]
```diff
@@ -1676,7 +1677,7 @@ class LeoPubsubMessageSubscriber[F[_]](
               AppError(
                 s"Error creating GCP app with id ${msg.appId} and cloudContext ${c.value}: DeleteAppV2 not supported for GCP",
                 ctx.now,
-                ErrorAction.CreateApp,
```
Comment (LizBaldo) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala:1679]: Wait, was that a bug?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala (hunk 1 above)
Comment (LizBaldo) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala:1074]: This is honestly a bug that I might have introduced everywhere. Might need to chat about it with the team during walkthrough?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala [hunk 6]
```diff
@@ -1295,26 +1281,80 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
...
+
+      // Update record in APP_CONTROLLED_RESOURCE table
+      _ <- appControlledResourceQuery
+        .updateStatus(
+          wsmResource.resourceId,
+          AppControlledResourceStatus.Deleting
+        )
+        .transaction
+
+      // Poll for databasee deletion
```
Comment (LizBaldo) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala]: nit `database`

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala [hunk 7]
```diff
@@ -1295,26 +1281,80 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
...
+
+      _ <- logger.info(ctx.loggingCtx)(s"WSM delete database job result for app ${app.appName.value}: $result")
+
+      _ <- F.raiseWhen(result.getJobReport.getStatus != JobReport.StatusEnum.SUCCEEDED)(
+        AppDeletionException(
+          s"WSM database deletion failed for app ${app.appName.value}. WSM response: $result"
+        )
+      )
+
+      // record in APP_CONTROLLED_RESOURCE table s updated by caller
```
Comment (LizBaldo) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala]: `is`

**Review by LizBaldo — APPROVED** [review]
LGTM! Only very minor comments since this is a noop given that most WSM delete endpoints are sync with the exception of the DB

Thanks for all of the tidying up!

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala (hunk 5 above)
Comment (jdcanas) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala:1679]: yep

**Pull request merged**